idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,659 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Compression Model.
+Reference:
+
+Modekurti et al., (2017). "Design, Dynamic Modeling, and
+Control of a Multistage CO2 Compresor System." International Journal
+of Greenhouse Gas Control. v62., page 31-45
+Created: April 2020
+__Author__ = "Quang Minh Le, John Eslick, Andrew Lee"
+"""
+
+from __future__ import division
+from pyomo.environ import (SolverFactory,
+                           NonNegativeReals, Var, Param,
+                           value, log, sqrt, exp,
+                           units as pyunits)
+from pyomo.common.config import ConfigValue, In
+from idaes.core import declare_process_block_class
+
+from idaes.generic_models.unit_models.pressure_changer import (
+    PressureChangerData, ThermodynamicAssumption)
+
+import idaes.logger as idaeslog
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.constants import Constants as const
+from enum import Enum
+
+_log = idaeslog.getLogger(__name__)
+
+
+class ImpellerType(Enum):
+    cover_impeller = 1
+    open_impeller = 2
+    custom = 3
+
+
+def _cover_impeller_callback(blk):
```

Comment:
Unless the user specifies the callback function by directly specifying a function, it's not really a callback.  You could incorporate this directly into build method, or if that is too messy just rename this like _build_covered_impeller_equations(), just to avoid confusion.